practice-two/src/App.jsx

```diff
@@ -1,18 +1,126 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 // Components
 import FilterCategoryLayout from './components/layout/FilterCategoryLayout';
 import ProductList from './components/layout/ListProductsCard';
 
-const App = () => (
-  <main className="m-auto p-3 max-w-[1300px] w-full min-w-[980px] gap-6">
-    <section className="pt-3 flex gap-4 relative">
-      <FilterCategoryLayout />
-    </section>
-    <section>
-      <ProductList />
-    </section>
-  </main>
-);
+// Data
+import LIST_PRODUCTS from '../database/products.json';
+
+/**
+ * Filters an array of products based on the selected manufacturer names.
+ * @param {Array} products - The array of products to filter.
+ * @param {Array} selectedFilter - An array of selected manufacture names to filter.
+ * @returns {Array} - An array containing the filtered products based on the selected manufacture.
+ */
+const filterProductsByManufacturer = (products, selectedFilter) => {
+  if (selectedFilter.length === 0) {
+    return products; // If no manufacturers are selected, return all products without filter
+  }
+
+  // Filter products based on whether their 'manufacturer'
+  return products.filter((product) => {
+    const selectedManufacturerObject = selectedFilter.map((manufacturer) => manufacturer.toLowerCase());
+    const productManufacturer = product.manufacturer.toLowerCase();
+
+    return selectedManufacturerObject.includes(productManufacturer);
+  });
+};
+
+/**
+ * Adds a 'manufacturer' property to each product.
+ * @param {Array} products - The array of products to add the 'manufacturer' property to.
+ * @returns {Array} - An array of products with the 'manufacturer' property added.
+ */
+const addManufacturerProperty = (products) => {
```

Comment:
what is this? I don't understand why you do it

Reply:
Removed